cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/DeleteScheduleStage.java

```diff
@@ -69,7 +70,9 @@ public void process(ApplicationWithPrograms input) throws Exception {
       scheduler.deleteSchedule(input.getApplicationId().program(programType,
                                                                 scheduleSpec.getProgram().getProgramName()),
                                scheduleSpec.getProgram().getProgramType(), scheduleSpec.getSchedule().getName());
-      programScheduler.deleteSchedule(input.getApplicationId().schedule(scheduleSpec.getSchedule().getName()));
```

Comment:
I can't comment on line 55 in this PR as there was no change. But: if a program is removed from the app, we need to delete its schedules, regardless of canUpdateSchedules().